src/main/java/seedu/uninurse/ui/CommandBox.java

```diff
@@ -82,4 +111,66 @@ public interface CommandExecutor {
         CommandResult execute(String commandText) throws CommandException, ParseException;
     }
 
+    /**
+     * The command history list.
+     */
+    private static final class CommandHistoryList {
+        private final int sizeLimit = 100;
+        private final List<String> history;
+        private int currentPointer;
+
+        /**
+         * Creates a {@code CommandHistoryList}.
+         */
+        public CommandHistoryList() {
+            history = new ArrayList<String>();
+            history.add("");
+            currentPointer = 0;
+        }
+
+        /**
+         * Adds a successful command to the history list, if the last command is a different one.
+         */
+        public void add(String command) {
+            if (history.size() < 2 || !history.get(history.size() - 2).equals(command)) {
+                // Update history only if the new command is different from the last one
+                history.set(history.size() - 1, command);
+                while (history.size() > sizeLimit) {
+                    history.remove(0);
+                }
+                history.add("");
+            }
+            currentPointer = history.size() - 1;
+        }
+
+        /**
+         * Handles the Up Key Button Pressed event.
+         */
+        public boolean handleUpKey() {
+            if (currentPointer == 0) {
+                return false;
+            }
+            currentPointer -= 1;
+            return true;
+        }
+
+        /**
+         * Handles the Down Key Button Pressed event.
+         */
+        public boolean handleDownKey() {
+            if (currentPointer + 1 == history.size()) {
+                return false;
+            }
+            currentPointer += 1;
+            return true;
+        }
+
+        /**
+         * Returns the command currently pointed to in the history list.
+         */
+        public String get() {
+            return history.get(currentPointer);
+        }
```

Comment:
I think you should let `handleUpKey` and `handleDownKey` return the current string instead of using the `get` method as now you are making an impure boolean method since it is modifying an internal variable and returning the status of the modification. 

Since `CommandHistoryList` is meant to return the command at the current pointer, the logic should be handled within itself when invoked by `handleUpKey` and `handleDownKey` respectively, rather than modifying an internal variable and verifying the status by an external agent (in this case it would be the `CommandBox` object) and retrieving another internal variable as now the logic is handled by an external agent. This introduces unnecessary coupling between these 2 classes (even though functionally it is not a big issue as `CommandHistoryList` is an internal class of `CommandBox`).

In summary, I think it would be better if you could update the implementation :D